src/relay.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Address,
+  BaseError,
+  createPublicClient,
+  createWalletClient,
+  ContractFunctionRevertedError,
+  getContract,
+  http,
+  WalletClient,
+} from "viem";
+
+import { celo, celoAlfajores } from "viem/chains";
+import config from "./config";
+import getSecret from "./get-secret";
+import { relayerAbi } from "./relayer-abi";
+import { privateKeyToAccount } from "viem/accounts";
+
+function getPublicClient() {
+  const isMainnet = process.env.NODE_ENV !== "development";
+  return createPublicClient({
+    chain: isMainnet ? celo : celoAlfajores,
+    transport: http(),
+  });
+}
+
+async function getWalletClient(): Promise<WalletClient> {
+  const isMainnet = process.env.NODE_ENV !== "development";
+  const pk = await getSecret(config.RELAYER_PK_SECRET_ID);
+
+  return createWalletClient({
+    account: privateKeyToAccount(pk as Address),
+    chain: isMainnet ? celo : celoAlfajores,
+    transport: http(),
+  });
+}
+
+export default async function relay(relayerAddress: string): Promise<boolean> {
+  const publicClient = getPublicClient();
+  const wallet = await getWalletClient();
+
+  const contract = getContract({
+    address: relayerAddress as Address,
+    abi: relayerAbi,
+    client: { public: publicClient, wallet },
+  });
+
+  try {
+    await contract.simulate.relay();
+
+    const hash = await contract.write.relay();
+    const receipt = await publicClient.waitForTransactionReceipt({
+      hash,
+      timeout: 50 * 1000, // 10 blocks for the tx to be mined
+    });
+
+    if (receipt.status !== "success") {
+      console.log(`Relay tx failed: ${hash}`);
+      return false;
+    }
+
+    console.log(`Relay tx succeeded: ${hash}`);
```

Comment:
given we might get multiple relay txs in parallel, should we include the rate feed name here for which we relayed?

same comment for line 57 above

Reply:
will do this as part of the logging refactoring